### PR TITLE
Remove reflection based flag generation

### LINKED ISSUE #5923: Remove reflection based flag generation
### Summary

Remove the last pieces of reflection based flag generation which are left after switching to `protoc-gen-go-flags`. 

### Current Situation

Currently the reflection based flag generation is still in use by the `simulate` command in the `cli`.

### Why do we need this? Who uses it, and when?

Clean up our technical debt there.

### Proposed Implementation

Generate flag definitions for the structs used in the `simulate` commands via `protoc-gen-go-flags` and remove the reflection based generation.

### Contributing

- [X] I can help by doing more research.
- [X] I can help by implementing the feature after the proposal above is approved.
- [X] I can help by testing the feature before it's released.

### Code of Conduct

- [X] I agree to follow TTN's [Community Code of Conduct](https://github.com/TheThingsNetwork/Manifesto/blob/master/CodeOfConduct.md).

### PR DESCRIPTION
**Pull request by ysmilda**
#### Summary

Closes #5923

#### Changes

- Remove reflection based flag generation

#### Testing

Manual testing. Limited to running `ttn-lw-cli simulate [command] --help` and checking if the flags were still generated correctly.

##### Regressions

n.a.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: cmd/ttn-lw-cli/internal/simulate/simulate_util.pb.go
Comment (adriansmares) [inline cmd/ttn-lw-cli/internal/simulate/simulate_util.pb.go]: I think the errors and `SetDefaults` should sit in a `simulate_util.go`. Otherwise if we regenerate this file we will override these functions.

**Review by adriansmares — APPROVED** [review]

File: cmd/ttn-lw-cli/internal/simulate/simulate_util.pb.go
Comment (ysmilda) [reply cmd/ttn-lw-cli/internal/simulate/simulate_util.pb.go]: Actuallly all the functions in this file are added manually, I'll rename it to `simulate.go`

**Comment by adriansmares** [comment]
What is the status here ?

**Comment by ysmilda** [comment]
Waiting for approval from @johanstokking so it can be merged.

**Review by johanstokking — APPROVED** [review]

**Comment by johanstokking** [comment]
Apologies, I thought @adriansmares's approval would lead to a green button but apparently not.

**Pull request merged**
